python/ray/serve/long_pull.py

```diff
@@ -0,0 +1,148 @@
+import asyncio
+import random
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Any, Awaitable, Callable, DefaultDict, Dict, Generic, Set, Tuple, List, TypeVar, Optional
+
+import ray
+from ray.serve.utils import logger
+
+
+@dataclass
+class UpdatedObject:
+    object_snapshot: Any
```

Comment:
Why is this called a snapshot? Seems like it's more of a "version," right? And the below is a version number?

Reply:
version number implies a sequential/temporal relationship. All we are doing here is some sort of identifier for the object value.